compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1748,8 +1748,8 @@ rustc_queries! {
         separate_provide_extern
     }
     /// Whether the function is an intrinsic
-    query is_intrinsic(def_id: DefId) -> bool {
-        desc { |tcx| "checking whether `{}` is an intrinsic", tcx.def_path_str(def_id) }
+    query intrinsic(def_id: DefId) -> Option<Symbol> {
```

Comment:
Would cg_clif need to use this too? It currently matches on the IntrinsicDef and uses `tcx.item_name()` when it is an intrinsic.

Reply:
`IntrinsicDef` is only created if `intrinsic` returns `Some`. Otherwise you get a normal definition. That's why I needed to query the backend. I created this design because I think it is more robust than letting the `IntrinsicDef` go all the way to backends and have them figure out how to call the fallback body again.

Basically if the backend doesn't say it has an impl for a rustc_intrinsic intrinsic, then the intrinsic becomes just a normal function from the perspective of the entire compiler

Reply:
The problem with the current design as far as I can tell is that the return value of `intrinsic` is hard coded in the crate metadata for libcore based on the codegen backend with which libcore is compiled. This would break linking crates compiled by different codegen backends together.

Reply:
Hmm... I guess I can make the backend check happen outside the query

Reply:
This is not relevant anymore. They are *always* intrinsics now, and the backends can choose to call the default body instead.